Tests/Approval.Tests/ApprovedApi/FluentAssertions/net47.verified.txt

```diff
@@ -2636,8 +2636,11 @@ namespace FluentAssertions.Xml
         public FluentAssertions.AndConstraint<FluentAssertions.Xml.XDocumentAssertions> BeEquivalentTo(System.Xml.Linq.XDocument expected, string because = "", params object[] becauseArgs) { }
         public FluentAssertions.AndWhichConstraint<FluentAssertions.Xml.XDocumentAssertions, System.Xml.Linq.XElement> HaveElement(string expected, string because = "", params object[] becauseArgs) { }
         public FluentAssertions.AndWhichConstraint<FluentAssertions.Xml.XDocumentAssertions, System.Xml.Linq.XElement> HaveElement(System.Xml.Linq.XName expected, string because = "", params object[] becauseArgs) { }
+        public FluentAssertions.AndWhichConstraint<FluentAssertions.Xml.XDocumentAssertions, System.Xml.Linq.XElement> HaveElement(string expected, FluentAssertions.OccurrenceConstraint occurrenceConstraint, string because = "", params object[] becauseArgs) { }
+        public FluentAssertions.AndWhichConstraint<FluentAssertions.Xml.XDocumentAssertions, System.Xml.Linq.XElement> HaveElement(System.Xml.Linq.XName expected, FluentAssertions.OccurrenceConstraint occurrenceConstraint, string because = "", params object[] becauseArgs) { }
```

Comment:
I think `AndWhichConstraint<XDocumentAssertions, IEnumerable<XElement>>` would be better for this overload.

Reply:
Hmm.. should HaveSingleElement() also return an `IEnumerable<XElement>`?

Comparing to the other `HaveElement()` methods.. they are returning an `AndWhichConstraint<XDocumentAssertions, XElement>`.

Reply:
This is what I imagined yesterday

```cs
AndWhichConstraint<XDocumentAssertions, IEnumerable<XElement>> HaveElement(string expected, OccurrenceConstraint occurrenceConstraint, ...);
AndWhichConstraint<XDocumentAssertions, IEnumerable<XElement>> HaveElement(XName expected, OccurrenceConstraint occurrenceConstraint, ...);

AndWhichConstraint<XDocumentAssertions, XElement> HaveSingleElement(string expected, ...);
AndWhichConstraint<XDocumentAssertions, XElement> HaveSingleElement(XName expected, ...);
```

but looking an extra time I now wonder if we should have `HaveSingleElement` at all? 
`HaveElement(expected)` is satisfied by one or more occurrences of `expected` and if you want to assert "one and only one" occurrence, use `HaveElement(expected, Exactly.Once())`.

Reply:
Yes 😎.. I have thought the same thing when I cherry-picked all commits from the initial author's feature branch.

And IMHO: `HaveSingleElement()` reads less "fluent" than `HaveElement("x", Exactly.Once());`